### PR TITLE
Update golinks.go

### PR DESCRIPTION
**Pull request by railroadkey**

### CODE REVIEW

**Comment by railroadkey** [comment]
numerous changes needed. 
-you don't need the for loop  ( i think go puts the http server in a go routine)
-looks like you don't need the exportable functions/values

**Comment by railroadkey** [comment]
see changes

File: golinks.go [hunk 1]
```diff
@@ -1,156 +1,160 @@
...
-	r.ReadConfig()
-
-	http.HandleFunc("/add/", r.AddLink)
-	http.HandleFunc("/list/", r.GetLinks)
-	http.HandleFunc("/del/", r.DelLink)
-	http.HandleFunc("/", r.Redirect)
-	if err := http.ListenAndServe(":"+*port, nil); err != nil {
-		log.Fatal("Unable to listen %s", err)
-	}
-	for {
```
Comment (railroadkey) [inline golinks.go:40]: for loop not needed

File: golinks.go [hunk 2]
```diff
@@ -1,156 +1,160 @@
...
-	Shortname string
-	Url       string
-	Requests  int32
+type redirect struct {
+        shortname string
+        url       string
+        requests  int32
 }
 
-type Settings struct {
```
Comment (railroadkey) [inline golinks.go:20]: everything works with non exported names

File: golinks.go [hunk 3]
```diff
@@ -1,156 +1,160 @@
...
+        if err != nil {
+                log.Printf("No config file found. Using new config file.")
+                return
+        }
+        if err := json.Unmarshal(jsonBlob, &m.redirects); err != nil {
+                log.Printf("Error unmarshalling %s", err)
+        }
 }
-
-func (m *Settings) SaveToDisk() {
```
Comment (railroadkey) [inline golinks.go:62]: this should probably return an error, then have the http handler provide an error message with http error code

File: golinks.go [hunk 4]
```diff
@@ -1,156 +1,160 @@
...
 
-func (m *Settings) ReadConfig() {
-	log.Printf("reading configuration...")
-	jsonBlob, err := ioutil.ReadFile(m.Filename)
-	if err != nil {
-		log.Printf("Unable to read configuration %s", err)
-		return
-	}
-	if err := json.Unmarshal(jsonBlob, &m.Redirects); err != nil {
-		log.Printf("error unmarshalling %s", err)
```
Comment (railroadkey) [inline golinks.go:58]: per go readability all errors are lower case, however logs are line oriented and need to capitalized. (hence i've capitalized all these log messages)

File: golinks.go [hunk 5]
```diff
@@ -1,156 +1,160 @@
...
+        return &settings{
+                filename: file,
+        }
 }
 
-func (m *Settings) ReadConfig() {
-	log.Printf("reading configuration...")
-	jsonBlob, err := ioutil.ReadFile(m.Filename)
-	if err != nil {
-		log.Printf("Unable to read configuration %s", err)
```
Comment (railroadkey) [inline golinks.go:54]: this error message confused me when i ran it the first time. how about something like "no config file found.. creating new config"

File: golinks.go [hunk 6]
```diff
@@ -1,156 +1,160 @@
...
+                }
+                // Add shortname, redirect to list.
+                m.redirects = append(m.redirects, &redirect{
+                        shortname: req[0],
+                        url:       "http://" + req[1],
+                        requests:  0,
+                })
+                err := m.saveToDisk()
+                if err != nil {
+                        http.Error(w, fmt.Sprintf("Internal error saving redirect."), http.StatusInternalServerError)
```
Comment (railroadkey) [inline golinks.go:142]: if disk save fails, provide error msg

**Comment by deepakkamesh** [comment]
updated. However I still needed exported members in struct for json marshal. Thanks for the comments.

**Pull request closed without merging**